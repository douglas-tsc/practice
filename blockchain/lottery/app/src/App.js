// @title Frontend for a lottery contract
// @author Josh

import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import web3 from './web3.js';
//   @dev this is a local file web3 not the web3 library

import lottery from './lottery.js';

class App extends Component {
  state = {
    manager: '',
    players: [],
    balance: '',
    value: 0
  };

  async componentDidMount() {
    const manager = await lottery.methods.manager().call();
    const players = await lottery.methods.getPlayers().call();
    const balance = await web3.eth.getBalance(lottery.options.address);
    this.setState({ manager, players, balance });
  }

  handleSubmit = async e => {
    e.preventDefault();
    const accounts = await web3.eth.getAccounts();
    this.setState({ message: 'Waiting on transaction success...' });
    await lottery.methods.enter().send({
      from: accounts[0],
      value: web3.utils.toWei(this.state.value, 'ether')
    });
    this.setState({
      message: 'You have been successfully entered into the lottery.'
    });
  };

  pickWinner = async () => {
    const accounts = await web3.eth.getAccounts();
    this.setState({ message: 'Waiting on transaction success...' });
    await lottery.methods.pickWinner().send({
      from: accounts[0]
    });
    this.setState({
      message: 'A winner has been picked.'
    });
  };

  render() {
    return (
      <div>
        <h1>Lottery Contract</h1>
        <p>This contract is managed by {this.state.manager}</p>
        <p>
          There are currently {this.state.players.length} players competing to
          win {web3.utils.fromWei(this.state.balance, 'ether')} ETH.
        </p>
        <hr />
        <form onSubmit={this.handleSubmit}>
          <h2>Want to try your luck?</h2>
          <label htmlFor="value">
            Amount of ether to enter{' '}
            <input
              type="number"
              value={this.state.value}
              onChange={e => this.setState({ value: e.target.value })}
            />
          </label>
          <input type="submit" value="Enter" />
        </form>
        <hr />
        <p>{this.state.message}</p>
        <hr />
        <p>Ready to pick a winner?</p>
        <button onClick={this.pickWinner}>Pick a Winner</button>
      </div>
    );
  }
}

export default App;

// @title A lottern contract
// @author Josh
//   @notice
/// @dev String comparison may be inefficient
/// @param _food The name of a food to evaluate (English)
/// @return true if Bugs will eat it, false otherwise
