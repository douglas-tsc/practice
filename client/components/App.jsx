import React from 'react';
import Header from './header.jsx';

export default (props) => {
  return (
    <div>
      <Header />
      {props.children}
    </div>
  );
};
