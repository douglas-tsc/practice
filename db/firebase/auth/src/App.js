import React, { Component } from 'react';
import SignIn from './SignIn';
import {auth} from './firebase';
import CurrentUser from './CurrentUser';
import Add from './Add';

class App extends Component {
  constructor (props) {
    super(props);
    this.state = {
      currentUser: null
    };
  }
  componentDidMount () {
    auth.onAuthStateChanged((currentUser) => {
      this.setState({currentUser});
    });
  }
  render () {
    return (
      <div className='flex x' >
        <div>
          <h2>Welcome to React</h2>
        </div>
        {this.state.currentUser ? <CurrentUser user={this.state.currentUser} /> : <SignIn />}
        <Add />
      </div>
    );
  }
}

export default App;
