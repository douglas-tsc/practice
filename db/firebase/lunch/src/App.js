import React, { Component } from 'react';
import SignIn from './SignIn';
import User from './User';
import NewRestaurant from './NewRestaurant';
import { auth, database } from './firebase';
import Restaurants from './Restaurants';

class App extends Component {
  constructor (props) {
    super(props);
    this.state = {
      currentUser: null,
      restaurants: null
    };
    this.restaurantsRef = database.ref('/restaurants');
  }

  componentDidMount () {
    auth.onAuthStateChanged((currentUser) => { this.setState({currentUser}); });

    this.restaurantsRef.on('value', (snapshot) => {
      this.setState({restaurants: snapshot.val()});
    });
  }
  render () {
    return (
      <div className='flex flex-column'>
        <div className='tc mv5'>
        <h1>Where should we eat?</h1>
          <div>
            ğŸ¥‘
          </div>
        </div>
        <div className='center w5'>
        {!this.state.currentUser && <SignIn />}
        {this.state.currentUser && <User user={this.state.currentUser} />}
        </div>
        <NewRestaurant />
        <Restaurants restaurants={this.state.restaurants}
          user={this.state.currentUser} />
      </div>
    );
  }
}

export default App;
