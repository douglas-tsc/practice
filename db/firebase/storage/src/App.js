import React, { Component } from 'react';
import { auth, database } from './firebase';
import CurrentUser from './CurrentUser';
import SignIn from './SignIn';
import ProfileCard from './ProfileCard';
import pick from 'lodash/pick' ;
import map from 'lodash/map';

class App extends Component {
  state = {
    user: null,
    users: {}
  }

  constructor (props) {
    super(props);
    this.userRef = null;
    this.usersRef = null;
  }

  componentDidMount () {
    auth.onAuthStateChanged((user) => {
      if (user) {
        this.setState({ user });

        this.usersRef = database.ref('/users');
        this.userRef = this.usersRef.child(user.uid);

        this.userRef.once('value').then((snapshot) => {
          if (snapshot.val()) return;
          const userData = pick(user, ['displayName', 'photoURL', 'email']);
          this.userRef.set(userData);
        });

        this.usersRef.on('value', (snapshot) => {
          this.setState({ users: snapshot.val() });
        });
      }
    });
  }
  render () {
    return (
      <div className=''>
        <div className=''>
          <h2>Welcome to React</h2>
        </div>
        {this.state.user
        ? <div>
          <CurrentUser
            user={this.state.user} />
        </div>
        : <SignIn />
      }
        <section>
          { map(this.state.users, (user, uid) => {
            return <ProfileCard key={uid} user={user} uid={uid} />;
          })}
        </section>
      </div>
    );
  }
}

export default App;
