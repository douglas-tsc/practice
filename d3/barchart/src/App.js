import React, { Component } from 'react'
import logo from './logo.svg'
import './App.css'
import * as d3 from 'd3'
import DynamicChart from './DynamicChart'
import DecimalChart from './DecimalChart'

class App extends Component {
  componentDidMount() {
    const data = [90, 70, 50, 30, 10]
    d3
      .select(`#chart`)
      .selectAll(`div`)
      .data(data)
      .enter()
      .append(`div`)
      .attr(`class`, `bar`)
      .style(`width`, d => `${d}px`)
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          <DecimalChart />
          <DynamicChart />
          <div id="chart" className="h5 w5 center" />
        </p>
      </div>
    )
  }
}

export default App
