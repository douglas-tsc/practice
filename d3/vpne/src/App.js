import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import logo from './logo.svg'
import './App.css'
import d3 from 'd3'

class App extends Component {
  constructor(props) {
    super(props)
    this.colors = d3.schemeCategory20
    this.width = d3.scaleBand().domain(d3.range(20))
  }

  componentDidMount() {
    ReactDOM.render(<Colors width="400" />, document.getElementById('svg'))
  }

  componentWillMount() {
    this.updateD3(this.props)
  }

  componentWillUpdate(newProps) {
    this.updateD3(newProps)
  }

  updateD3(props) {
    this.width.range([0, props.width])
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <g>
          {d3
            .range(20)
            .map(i =>
              <Swatch
                color={this.colors[i]}
                width={this.width.step()}
                x={this.width(i)}
                y="0"
              />
            )}
        </g>
      </div>
    )
  }
}

const Swatch = ({ color, width, x }) =>
  <rect width={width} height="20" x={x} y="0" style={{ fill: color }} />

export default App
