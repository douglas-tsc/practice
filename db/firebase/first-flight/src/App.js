import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import {database} from './firebase';

class App extends Component {
  constructor (props) {
    super(props);
    this.state = {
      data: null,
      newData: ''
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  componentDidMount () {
    database.ref('/firstFlight').on('value', (snapshot) => {
      snapshot.key;
      this.setState({
        data: snapshot.val()
      });
    });
  }
  handleChange (e) {
    const newData = e.target.value;
    this.setState({ newData });
  }
  handleSubmit (e) {
    e.preventDefault();
    const newData = database.ref()
    .child('firstFlight')
    .set(this.state.newData);
  }

  render () {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to React & Firebase</h2>
        </div>
        <p className='App-intro'>
          {this.state.data || '...'} from firebase</p>
        <form className='flex col mw5 center' onSubmit={this.handleSubmit}>
          <input type='text' onChange={this.handleChange}
            value={this.state.newData} />
          <input type='submit' value='submit' />
        </form>
      </div>
    );
  }
}

export default App;
