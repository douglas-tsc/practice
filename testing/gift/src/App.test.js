import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
})

const app = shallow(<App />)

it('renders correctly', () => {
  expect(app).toMatchSnapshot()
})

it(`initializes state with empty array of gifts`, () => {
  expect(app.state().gifts).toEqual([])
})

it(`adds a new gift to state when clicking on 'add gift' button`, () => {
  app.find(`[data-test='addButton']`).simulate('click')
  expect(app.state().gifts).toEqual([{ id: 1 }])
})
