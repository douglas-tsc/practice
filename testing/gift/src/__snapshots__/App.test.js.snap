// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <h1
            className="App-title"
          >
            Gift Giver
          </h1>
        </header>,
        <button
          data-test="addButton"
          onClick={[Function]}
        >
          Add
        </button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <h1
              className="App-title"
            >
              Gift Giver
            </h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Gift Giver",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Gift Giver",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
          "data-test": "addButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h1
              className="App-title"
            >
              Gift Giver
            </h1>
          </header>,
          <button
            data-test="addButton"
            onClick={[Function]}
          >
            Add
          </button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <h1
                className="App-title"
              >
                Gift Giver
              </h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Gift Giver",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Gift Giver",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "data-test": "addButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
